### DISCUSSION TITLE
How to correctly use a SQL DB with NextJS (without express)

### DISCUSSION THREAD

**Discussion by Sillyvan** [vercel/next.js#15183] [Help] [answered]
Hi!

Im pretty new with Next and im a little bit confused.
Im using the /api folder like in the create-next-app default example and i want to get data from my DB.
Now i used the MariaDB driver and im creating a pool for the connections. i dont think this works correctly so thats why im asking..

Im getting the data and everything runs fine but! i think everytime someone fetches the API the driver creates a new connection pool. Is this because Next works "serverless"? i saw a doc by vercel that used serverless-mysql. so im wondering do i need a serverless driver? or can i just use a "normal" connection instead of using a pool? or can i use a connection pool?

This is my first project where im making the API in next so any tips or help in that regard is highly appriciated.
Silvan

**Comment by leerob** [comment, marked as answer]
The Next.js + MySQL example has been updated, including a brand new demo 🎉

Code: https://github.com/vercel/next.js/tree/canary/examples/with-mysql
Demo: https://next-mysql.vercel.app/

**Reply by ShawnCentra** [reply]
Does this work with SQL servers too?

**Reply by leerob** [reply]
Sure!

**Comment by schizek** [comment]
@leerob are there any examples that use actual MySQL with real database queries (i.e. `SELECT` statements and such) instead of being abstracted away in yet another 3rd party package or is Prisma necessary to use MySQL with Next.js?

**Reply by schizek** [reply]
FWIW... I ended up using [serverless/mysql](https://github.com/jeremydaly/serverless-mysql) and it appears to be working and I'm no longer getting excessive connections.
